front/app/components/IdeaCards/SortFilterDropdown.tsx

```diff
@@ -10,63 +10,79 @@ import {
   IdeaDefaultSortMethod,
   ideaDefaultSortMethodFallback,
 } from 'services/participationContexts';
+import { IProjectData } from 'services/projects';
+import { TPhase } from 'hooks/usePhase';
+import { getMethodConfig } from 'utils/participationMethodUtils';
+import { isNilOrError } from 'utils/helperUtils';
 
 type Props = {
   id?: string | undefined;
   alignment: 'left' | 'right';
   onChange: (value: string) => void;
   defaultSortingMethod?: IdeaDefaultSortMethod;
+  phase?: TPhase | null;
+  project?: Error | IProjectData | null;
 };
 
-type State = {
-  selectedValue: string[];
-};
+const SortFilterDropdown = ({
+  alignment,
+  onChange,
+  defaultSortingMethod,
+  phase,
+  project,
+}: Props) => {
+  const [selectedValue, setSelectedValue] = useState<string[]>([
+    defaultSortingMethod || ideaDefaultSortMethodFallback,
+  ]);
 
-class SortFilterDropdown extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      selectedValue: [
-        props.defaultSortingMethod || ideaDefaultSortMethodFallback,
-      ],
-    };
-  }
+  useEffect(() => {
+    if (defaultSortingMethod) {
+      setSelectedValue([defaultSortingMethod]);
+    }
+  }, [defaultSortingMethod]);
 
-  handleOnChange = (selectedValue: string[]) => {
-    this.setState({ selectedValue });
-    this.props.onChange(selectedValue[0]);
+  const handleOnChange = (selectedValue: string[]) => {
+    setSelectedValue([selectedValue[0]]);
+    onChange(selectedValue[0]);
   };
 
-  sortTitle = (<FormattedMessage {...messages.sortTitle} />);
+  const sortTitle = <FormattedMessage {...messages.sortTitle} />;
 
-  render() {
-    const { alignment } = this.props;
-    const { selectedValue } = this.state;
-    const options = [
-      { text: <FormattedMessage {...messages.trending} />, value: 'trending' },
-      { text: <FormattedMessage {...messages.random} />, value: 'random' },
-      { text: <FormattedMessage {...messages.popular} />, value: 'popular' },
-      { text: <FormattedMessage {...messages.newest} />, value: 'new' },
-      { text: <FormattedMessage {...messages.oldest} />, value: '-new' },
-    ];
+  let options = [
```

Comment:
This piece of code doesn't depend on any application state and could be defined as a constant outside of the component. In my opinion doing it like that is cleaner conceptually (subjective), but also more performant (object only needs to be created once instead of on every re-render) and could avoid bugs (the array passed down to the `FilterSelector` will be a stable reference)

Reply:
Actually, this exact same array seems to be reused a bunch of times, like in `IdeationInputs` and `participationMethodUtils`. Maybe we could just define it once and then import it and reuse it? Maybe `participationMethodUtils` would be a logical place for it to live?

Reply:
Ah yes you're completely right - I'll add it to the `participationMethodUtils` :) Thanks!!